tools/SourceKit/include/SourceKit/Support/Concurrency.h

```diff
@@ -17,6 +17,55 @@
 
 namespace SourceKit {
 
+class Semaphore {
+public:
+  Semaphore(long count) {
+    ImplObj = Impl::create(count);
+  }
+  ~Semaphore() {
+    if (ImplObj)
+      Impl::release(ImplObj);
+  }
+
+  void signal() {
+    Impl::signal(ImplObj);
+  }
+
+  bool wait() {
+    return Impl::wait(ImplObj);
+  }
+
+  bool wait(long milliseconds) {
+    return Impl::wait(ImplObj, milliseconds);
+  }
+
+  Semaphore(const Semaphore &Other) {
```

Comment:
Having semaphores be copy-constructible sounds like a bad idea to me.  We should be passing them by reference.  What do you think?